watchtower-plugin/src/retrier.rs

```diff
@@ -727,24 +734,24 @@ mod tests {
             .add_pending_appointment(tower_id, &appointment);
 
         // Start the task and send the tower to the channel for retry
-        let wt_client_clone = wt_client.clone();
-
-        let mut retry_manager = RetryManager::new(
-            wt_client_clone,
-            rx,
-            MAX_ELAPSED_TIME + 1,
-            SHORT_AUTO_RETRY_DELAY,
-            MAX_INTERVAL_TIME,
-        );
-        let task = tokio::spawn(async move { retry_manager.manage_retry().await });
         tx.send((tower_id, RevocationData::Fresh(appointment.locator)))
             .unwrap();
 
-        // Wait for the elapsed time and check how the tower status changed
-        tokio::time::sleep(Duration::from_secs_f64(
-            (MAX_ELAPSED_TIME as f64 + 1.0) / 2.0,
-        ))
-        .await;
+        let wt_client_clone = wt_client.clone();
+        let task = tokio::spawn(async move {
+            RetryManager::new(
+                wt_client_clone,
+                rx,
+                MAX_ELAPSED_TIME,
+                SHORT_AUTO_RETRY_DELAY,
+                MAX_INTERVAL_TIME,
+            )
+            .manage_retry()
+            .await
+        });
+
+        // Wait for one retry round and check to tower status
+        tokio::time::sleep(Duration::from_secs_f64(MAX_RUN_TIME)).await;
```

Comment:
> ```rust
> // Wait for one retry round and check to tower status
> 754|tokio::time::sleep(Duration::from_secs_f64(MAX_RUN_TIME)).await;
> assert!(temp_unreachable);
> assert!(running);
> // Wait until the task gives up and check again (this gives up due to accumulation of transient errors,
> // so the retiers will be idle).
> 770|tokio::time::sleep(Duration::from_secs(MAX_ELAPSED_TIME as u64)).await;
> assert!(unreachable);
> assert!(idle);
> ```
> 
> nit: We can make the second sleep (line 770) be `MAX_ELAPSED_TIME + MAX_RUN_TIME` instead of `MAX_ELAPSED_TIME` for clarity. As this accounts for an extra `MAX_RUN_TIME` (last `self.run` round) after `MAX_ELAPSED_TIME` have passed. But it would be correct either way since L754 already waits `MAX_RUN_TIME`.

I think I'll add a comment there specifying that we are not adding it due to it had been accounted for right before, so we don't have to add an unnecessary wait (not that they make sense haha)